spring-cloud-stream/src/main/java/org/springframework/cloud/stream/endpoint/BindingsEndpoint.java

```diff
@@ -0,0 +1,105 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.stream.endpoint;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import org.springframework.beans.DirectFieldAccessor;
+import org.springframework.boot.actuate.endpoint.annotation.Endpoint;
+import org.springframework.boot.actuate.endpoint.annotation.ReadOperation;
+import org.springframework.boot.actuate.endpoint.annotation.WriteOperation;
+import org.springframework.cloud.stream.binder.Binding;
+import org.springframework.cloud.stream.binding.InputBindingLifecycle;
+import org.springframework.cloud.stream.config.BindingServiceProperties;
+import org.springframework.util.CollectionUtils;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+@Endpoint(id = "bindings")
+public class BindingsEndpoint {
+
+	private final List<InputBindingLifecycle> inputBindingLifecycles;
+	
+	public BindingsEndpoint(List<InputBindingLifecycle> inputBindingLifecycles, BindingServiceProperties properties) {
+		this.inputBindingLifecycles = inputBindingLifecycles;
+	}
+	
+	
+	@ReadOperation
+	public List<Object> bindings() {
+		List<Binding<Object>> inputBindings = this.gatherInputBindings();
+		
+		return new ObjectMapper().convertValue(inputBindings, new TypeReference<List<Object>>() {});	
+	}
+	
+	public StopEndpoint getStopEndpoint() {
+		return new StopEndpoint();
+	}
+	
+	public StartEndpoint getStartEndpoint() {
+		return new StartEndpoint();
+	}
+	
+	@SuppressWarnings("unchecked")
+	private List<Binding<Object>> gatherInputBindings() {
+		List<Binding<Object>> inputBindings = new ArrayList<>();
+
+		for (InputBindingLifecycle inputBindingLifecycle : inputBindingLifecycles) {
+			DirectFieldAccessor dfa = new DirectFieldAccessor(inputBindingLifecycle);
+			Collection<Binding<Object>> lifecycleInputBindings = (Collection<Binding<Object>>) dfa.getPropertyValue("inputBindings");
+			if (!CollectionUtils.isEmpty(lifecycleInputBindings)) {
+				inputBindings.addAll(lifecycleInputBindings);
+			}	
+		}
+		return inputBindings;
+	}
+	
+	@Endpoint(id = "bindings/stop/{name}")
+	public class StopEndpoint {	
+		@WriteOperation
+		public boolean stop(String name) {
+			List<Binding<Object>> inputBindings = BindingsEndpoint.this.gatherInputBindings();
+			
+			for (Binding<Object> binding : inputBindings) {
+				String bindingName = binding.getName();
+				if (name.equals(bindingName)) {
+					binding.unbind();
```

Comment:
Maybe we should `return true` exactly here (not sure if we can have more than one binding with that name) and `false` in the end of method to indicate that there is nothing to stop?

Also how about to consider to implement this logic via Java `Stream` ?

Reply:
sure, once the approach is settled on